build/pkgs/singular/spkg-configure.m4

```diff
@@ -6,14 +6,27 @@ SAGE_SPKG_CONFIGURE([singular], [
     AS_IF([test -z "${SINGULAR_BIN}"], [sage_spkg_install_singular=yes], [
       dnl Use pkg-config to ensure that Singular is new enough.
       PKG_CHECK_MODULES([SINGULAR], [Singular >= 4.2.1], [
-       AC_MSG_CHECKING([that Singular's help is working])
-       AS_IF([test x`printf "system(\"--browser\", \"builtin\"); \n help;" | Singular 2>&1 | grep "error occurred"` = x], [
-        AC_MSG_RESULT(yes)
-       ], [
-       AC_MSG_RESULT(no)
-       sage_spkg_install_singular=yes
-       ]
-      )], [
+        AC_MSG_CHECKING([whether Singular is built with FLINT])
+        AC_COMPILE_IFELSE([
+          AC_LANG_PROGRAM([
+            #include <singular/singularconfig.h>
+            #if !defined(HAVE_FLINT)
+            #  error "Need Singular compiled with FLINT"
```

Comment:
Is there any chance that this program compiles?

It seems to be missing `#endif` and a `main(){}` although I'm not completely sure on the semantics of `AC_LANG_PROGRAM`. In any case, this test fails in void linux where singular is built with flint (and `/usr/include/singular/singularconfig.h` indeed does contain `#define HAVE_FLINT 1`)

Reply:
Fixed in #36212, I think